src/main/java/com/ericsson/ei/notifications/HttpRequest.java

```diff
@@ -183,11 +193,12 @@ private void addAuthenticationData() throws AuthenticationException {
      */
     private boolean isAuthenticationDetailsProvided(String authType, String username,
             String password) {
-        if (authType.isEmpty() || authType.equals(AUTHENTICATION_TYPE_NO_AUTH)) {
+
+        if (authType.isEmpty() || authType.equals(AuthenticationType.NO_AUTH.getValue())) {
             return false;
         }
 
-        if (username.isEmpty() && password.isEmpty()) {
+        if (username.isEmpty() || password.isEmpty()) {
```

Comment:
If using token, then that is stored as password but username is empty. That is why it was && beforew.

Reply:
Ok I see